src/manifold/src/sort.cpp

```diff
@@ -432,4 +496,138 @@ void Manifold::Impl::GatherFaces(const Impl& old,
                           old.halfedge_.cptrD(), old.halfedgeTangent_.cptrD(),
                           faceNew2Old.cptrD(), faceOld2New.cptrD()}));
 }
+
+/// Constructs a position-only MeshGL from the input Mesh.
+MeshGL::MeshGL(const Mesh& mesh) {
+  numProp = 3;
+  vertProperties.resize(numProp * mesh.vertPos.size());
+  for (int i = 0; i < mesh.vertPos.size(); ++i) {
+    for (int j : {0, 1, 2}) vertProperties[3 * i + j] = mesh.vertPos[i][j];
+  }
+  triVerts.resize(3 * mesh.triVerts.size());
+  for (int i = 0; i < mesh.triVerts.size(); ++i) {
+    for (int j : {0, 1, 2}) triVerts[3 * i + j] = mesh.triVerts[i][j];
+  }
+  halfedgeTangent.resize(4 * mesh.halfedgeTangent.size());
+  for (int i = 0; i < mesh.halfedgeTangent.size(); ++i) {
+    for (int j : {0, 1, 2, 3})
+      halfedgeTangent[4 * i + j] = mesh.halfedgeTangent[i][j];
+  }
+}
+
+/**
+ * Updates the mergeFromVert and mergeToVert vectors in order to create a
+ * manifold solid. If the MeshGL is already manifold, no change will occur and
+ * the function will return false. Otherwise, this will merge verts along open
+ * edges within precision (the maximum of the MeshGL precision and the baseline
+ * bounding-box precision), keeping any from the existing merge vectors.
+ *
+ * There is no guarantee the result will be manifold - this is a best-effort
+ * helper function designed primarily to aid in the case where a manifold
+ * multi-material MeshGL was produced, but its merge vectors were lost due to a
+ * round-trip through a file format.
+ */
+bool MeshGL::Merge() {
+  std::multiset<Edge> openEdges;
+
+  std::vector<int> merge(NumVert());
```

Comment:
It seems to me that when `mergeFromVert.size()` is much smaller than `NumVert()`, it may be more beneficial to use a sparse hashmap. (just a nitpick)

Reply:
Well, it might save a bit of memory, but not compute. I figure if I'm not allocating more memory than I already have stored in the structure I'm operating on, then the savings probably isn't worth the trouble.

Reply:
It depends on the size and sparsity. For large data set, saving memory equals to better cache utilization which means better performance. But yeah this may not worth the trouble.